src/daft-parquet/src/read.rs

```diff
@@ -22,18 +24,47 @@ use snafu::ResultExt;
 
 use crate::{file::ParquetReaderBuilder, JoinSnafu};
 
+#[cfg(feature = "python")]
+#[derive(Default, Clone)]
+pub struct ParquetSchemaInferenceOptionsBuilder {
+    pub coerce_int96_timestamp_unit: Option<PyTimeUnit>,
+    pub coerce_string_to_binary: Option<bool>,
+}
```

Comment:
Why do we need a builder? Could we not just implement a `ParquetSchemaInferenceOptions::new()`?